velox/core/PlanNode.h

```diff
@@ -1768,6 +1768,74 @@ class MergeJoinNode : public AbstractJoinNode {
   static PlanNodePtr create(const folly::dynamic& obj, void* context);
 };
 
+struct IndexJoinCondition;
+using IndexJoinConditionPtr = std::shared_ptr<IndexJoinCondition>;
+struct IndexJoinCondition : public ISerializable {
```

Comment:
Just my 2 cents: Does `IndexJoinCondition` have to extend `ISerializable` given the subclasses are unlikely using "name" JSON field as type annotations?